src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -559,6 +562,40 @@ private static SyntaxNode GetTopMostParentingExpression(ExpressionSyntax express
                     newNode.WithAdditionalAnnotations(ConflictAnnotation.Create(CSharpFeaturesResources.Conflict_s_detected));
 
             return await inlinedDocument.ReplaceNodesAsync(replacementNodesWithChangedSemantics.Keys, conflictAnnotationAdder, cancellationToken).ConfigureAwait(false);
+
+            void initReplacementNodesWithChangedSemantics()
```

Comment:
this will haven allocation because of the capture.  at that poiint, we're not gaining anything.  either change the code to

```replacementNodes = replacementNodes ?? new ...```

Or just initialize up front and check for empty instead of null at the bottom.

Reply:
Hum, I'm not sure. If I recall correctly, local functions use structs for the closure.

Reply:
@CyrusNajmabadi Does that make sense?

Reply:
Oh, maybe they do.  i imagine @agocke would know the most.  Regardless, i think this is a bit of overkill in terms of the 'unify code' pattern.  it think just having ```replacementNodes = replacementNodes ?? new ...``` would be fine :)

Reply:
@CyrusNajmabadi Done ;-)

Reply:
I agree with @CyrusNajmabadi that saving one line isn't really worth a method, but no, this won't allocate because the local function is not converted to a delegate.

Reply:
@agocke that's very nifty.  I knew we were thinking about htat optimization.  I didn't realize we had done it!